### ISSUE TITLE
[REST API] Empty "binding" property will cause malformed error when creating/updating package

### ISSUE THREAD

**Issue by archerzz** [openwhisk/openwhisk#1368]
## Environment details:
- REST API
## Steps to reproduce the issue:
1.   Send a PUT request to create or update a package
2.   In the JSON object, specify "binding" with "{}".
## Provide the expected results and outputs:

```
Package is successfully created or updated.
```
## Provide the actual results and outputs:

```
Error message is returned:
The request content was malformed: Object is missing required member 'namespace'
```
## Additional information you deem important:
- It's quite common that JSON library will create empty object when the property is null. At least we should accept an empty object.
- This error seems related to creation of a bind. A bind action re-uses the same REST API.

**Comment by rabbah** [comment]
The deserialization of `Binding` from a PUT request is done here: https://github.com/openwhisk/openwhisk/blob/master/common/scala/src/main/scala/whisk/core/entity/WhiskPackage.scala#L192. We would have to change this to be a custom deserializer like the one used on line 143.

**Comment by psuter** [comment]
Fixed as of c906bd9c8f83055ce9a5b1f6eee6f411ed161460
